packages/commonwealth/server/routes/comments/update_comment_handler.ts

```diff
@@ -25,8 +26,23 @@ export const updateCommentHandler = async (
   res: TypedResponse<UpdateCommentResponse>
 ) => {
   const { user, chain, address } = req;
-  const { id: commentId } = req.params;
-  const { body: commentBody } = req.body;
+  const { id } = req.params;
+  const { body: commentBody, discord_meta } = req.body;
+
+  let commentId = id;
+
+  // Special handling for discobot threads
+  if (discord_meta !== undefined && discord_meta !== null) {
+    const existingComment = await controllers.threads.models.Comment.findOne({
+      where: { discord_meta: discord_meta },
+    });
+    if (existingComment) {
+      commentId = existingComment.id;
+    } else {
+      throw new AppError(Errors.NoId);
+    }
```

Comment:
Because of how the discord bot works, we don't have access to the comment Id directly when we want to update it- hence this logic. The updateComment method expects that commentId. How do you want me to handle that discrepancy? Can I make an another function in update_comment.ts that will handle this logic before calling updateComment? Seems like we should be able to avoid rewriting updateComment just to accommodate this edge case. @rbennettcw

Reply:
Feel free to update the controller methods and types however is needed

In this case we need to make `commentId` optional. We can remove this check from the handler:
```ts
  if (!commentId) {
    throw new AppError(Errors.NoId);
  }
```

The new business logic can be moved to the `update_comment.ts`, and the `UpdateCommentOptions` type can make `commentId` optional

Some unit tests may need to be updated to accommodate the change

Reply:
makes sense